test/utils/skipIf.ts

```diff
@@ -0,0 +1,29 @@
+// Shim for vitest describe.skipIf to be able to run mocha and vitest side-by-side
+/**
+ * Skip a test suite if a condition is met.
+ * @param {boolean} condition - The condition to check.
+ * @returns {Function} The test suite function.
+ */
+const describeSkipIf: (condition: boolean) => Mocha.PendingSuiteFunction =
+  (describe as any).skipIf ??
+  function describeSkipIf(condition: boolean) {
+    return condition ? describe.skip : describe;
+  };
+
+/**
+ * Skip a test if a condition is met.
+ * @param {boolean} condition - The condition to check.
+ * @returns {Function} The test function.
+ */
+const testSkipIf: (condition: boolean) => Mocha.PendingTestFunction =
+  (it as any).skipIf ??
+  function testSkipIf(condition: boolean) {
+    return condition ? it.skip : it;
+  };
+
+const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+const isOSX = /macintosh/i.test(window.navigator.userAgent);
+const hasTouchSupport =
+  typeof window.Touch !== 'undefined' && typeof window.TouchEvent !== 'undefined';
+
+export { describeSkipIf, testSkipIf, isJSDOM, isOSX, hasTouchSupport };
```

Comment:
Instead of `@mui/internal-test-utils` as it is easier to maintain. We can merge once the pattern is settled, should be as easy as fixing the imports. Or removing it altogether if we go that way.